bin/vim_erlang_tags.erl

```diff
@@ -120,77 +120,126 @@ get_command_type(C) when C=:=include; C=:=ignore; C=:=output -> stateful;
+    EtsTags = create_tags(Explore),
+    ok = tags_to_file(EtsTags, TagFile),
+    ets:delete(EtsTags).
+
+set_verbose_flag(#{verbose := T}) ->
+    put(verbose, T),
+    log("Verbose mode on.~n").
+
+-spec reparse_args(parsed_params(), cmd_line_arguments()) -> parsed_params().
+reparse_args(Opts, []) ->
+    Opts;
+reparse_args(Opts, AllArgs) ->
+    {Param, ToContinueParsing} = parse_next_arg(AllArgs),
+    {ParamState, NextArgs} = case get_command_type(Param) of
+        boolean -> {true, ToContinueParsing};
+        stateful -> get_full_arg_state(Param, maps:get(Param, Opts, []), ToContinueParsing)
+    end,
+    reparse_args(Opts#{Param := ParamState}, NextArgs).
+
+-spec parse_next_arg(nonempty_list(cmd_line_arg())) -> {cmd_param(), cmd_line_arguments()}.
+parse_next_arg([Arg | NextArgs] = AllArgs) ->
+    lists:foldl(
+      fun({Param, ParamList}, Acc) ->
+              case lists:member(Arg, ParamList) of
+                  true -> {Param, NextArgs};
+                  _ -> Acc
+              end
+      end,
+      {include, AllArgs}, %% If the parameter is not recognised, just throw it into include
+      allowed_cmd_params()).
+
+%% Return args for the current parameter, and the rest of the args to continue parsing
+-spec get_full_arg_state(Param, CurrentParamState, ToContinueParsing) -> Ret when
+    Param :: cmd_param(),
+    CurrentParamState :: cmd_line_arguments(),
+    ToContinueParsing :: cmd_line_arguments(),
+    Ret :: {boolean(), cmd_line_arguments()}
+         | {cmd_line_arguments(), cmd_line_arguments()}.
+get_full_arg_state(Param, _CurrentParamState, ToContinueParsing)
+  when Param =:= otp; Param =:= help; Param =:= verbose ->
+    {true, ToContinueParsing};
+get_full_arg_state(Param, CurrentParamState, ToContinueParsing) ->
+    log("Parsing args for parameter ~p~n", [Param]),
+    {StateArgs, Rest} = consume_until_new_state(ToContinueParsing),
+    case StateArgs of
+        [] -> log_error("Arguments needed for ~s.~n", [Param]);
+        _ -> ok
     end,
-    put(ignored, AllIgnored),
-    parse_args(OtherArgs);
-parse_args(["-" ++ Arg|_]) ->
-    log_error("Unknown argument: ~s~n", [Arg]),
-    halt(1);
-parse_args([FileName|OtherArgs]) ->
-    put(files, [FileName|get(files)]),
-    parse_args(OtherArgs).
+    {StateArgs ++ CurrentParamState, Rest}.
+
+-spec consume_until_new_state(Args) -> {ConsumedArgs, RestArgs} when
+      Args :: cmd_line_arguments(),
+      ConsumedArgs :: cmd_line_arguments(),
+      RestArgs :: cmd_line_arguments().
+consume_until_new_state(Args) ->
+    log("    Consuming args ~p~n", [Args]),
+    States = lists:foldl(fun({_,S}, Acc) -> S ++ Acc end, [], allowed_cmd_params()),
+    lists:splitwith(
+      fun("-" ++ _ = El) ->
+              case lists:member(El, States) of
+                  true -> false;
+                  _ -> log_error("Unknown argument: ~s~n", [El]), halt(1)
+              end;
+         (El) ->
+              not lists:member(El, States)
```

Comment:
Is it possible that `El` is a member of the `States` list? If not, then simply `true` could suffice.